File: apps/api/src/contexts/core/domain/repositories/event_repository.py
from abc import ABC, abstractmethod

from returns.result import Result

from ..entities import Event
from ..value_objects.event_id import EventId


class EventRepository(ABC):
    @abstractmethod
    def create(self, event: Event) -> Result[None, Exception]:
        pass

    @abstractmethod
    def persist(self, event: Event) -> Result[None, Exception]:
        pass

    @abstractmethod
    def get(self, event_id: EventId) -> Result[Event | None, Exception]:
        pass

File: apps/tickets/src/contexts/core/domain/repositories/event.repository.ts
import { Result } from 'neverthrow';
import { Event } from '../entities/event';
import { EventId } from '../value-objects/event-id';

export abstract class EventRepository {
  abstract save(event: Event): Promise<Result<void, Error>>;
  abstract delete(eventId: EventId): Promise<Result<void, Error>>;
  abstract findById(eventId: EventId): Promise<Result<Event | null, Error>>;
}

